File: LeetCodeOJ/8.cpp
long long StrToInt(char *digit, int minus){
    long long num = 0;
    while(*digit != '\0'){
        if(*digit >= '0' && *digit <= '9'){
            int flag = minus ? -1 : 1;
            num = num * 10 + flag * (*digit - '0');
            if(!minus && num > 0x7FFFFFFF){
                num = 0x7FFFFFFF;
                break;
            }
            else if(minus && num < (signed int)0x80000000){
                num = 0x80000000;
                break;
            }
            digit++;
        }
        else{
            break;
        }
    }
    return num;
}

int myAtoi(char* str) {
    long long num = 0;
    int minus = 0;
    if(str != NULL && *str != '\0'){
        char *digit = str;
        int flag = 0;
        
        while(*digit != '\0'){
            if(*digit == ' '){
                if(flag == 0){
                    digit++;
                }
                else{
                    break;
                }
            }
            else if(*digit == '+'){
                if(flag == 0){
                    flag = 1;
                    digit++;
                }
                else{
                    break;
                }
            }
            else if(*digit == '-'){
                if(flag == 0){
                    minus = 1;
                    flag = 1;
                    digit++;
                }
                else{
                    break;
                }
            }
            else if(*digit >= '0' && *digit <= '9'){
                num = StrToInt(digit, minus);
                break;
            }
            else{
                break;
            }
        }
    }
    return (int)num;
}

File: LeetCodeOJ/8.py
class Solution(object):
    def myAtoi(self, str):
        """
        :type str: str
        :rtype: int
        """
        self.s = str.strip()
        num = 0
        if self.s != None and len(self.s) != 0:
            flag = 0
            minus = 1
            for i in self.s:
                if i == "+":
                    if flag == 0:
                        flag = 1
                    else:
                        break
                elif i == "-":
                    if flag == 0:
                        flag = 1
                        minus = -1
                    else:
                        break
                elif i >= '0' and i <= '9':
                    flag = 1
                    num = num * 10 + minus * int(i)
                else:
                    break
            
            if minus == 1 and num > 0x7FFFFFFF:
                num = 0x7FFFFFFF
            elif minus == -1 and (num < -0x80000000):
                num = -0x80000000
        return num
                    
        
